prov/psm/src/psmx_init.c

```diff
@@ -398,7 +399,7 @@ PSM_INI
 	int check_version;
 	int err;
 
-	fi_log_init();
+	psm_handle = fi_log_register_provider("psm");
```

Comment:
If we're making logging always available, logging can be initialized automatically by the framework.  I think the framework can set this up in fi_register_provider.